### ISSUE TITLE
VPN killswitch

### ISSUE THREAD

**Issue by augugusto** [guardianproject/orbot#954]
I like to use orbot con malls and similar to stop the tracking and spying. However, I always have to first connect to the WiFi, then enable orbot. On most cases, when you regain internet access, a lot of apps start sending pending data, and pulling new one, so those first seconds are the most important. 

I know android has a way to do this, but requires me going through settings and is a pain in general.

I'd like a kill switch option on the app itself

**Comment by SkewedZeppelin** [comment]
> first connect to the WiFi

Did you disable your Internet connectivity checks?

> android has a way to do this

It is far more robust than what an app can implement.

**Comment by tkbstudios** [comment]
This isn't possible to add in-app to my knowledge or is difficult to add, just go to your VPN settings on Android and enable always-on

**Comment by augugusto** [comment]
It's not impossible. In fact, I believe it working right now. I don't know of it works because of a bug or because of a feature, but I think it is. 
Even if it  wasn't working it still wouldn't be hard. Before, the app marked the VPN as connected only when actually connected. What they would have to do is to always report the VPN as connected (so that android uses it to send all packages), and then if the connection is not actually established, drop the packet

**Comment by tkbstudios** [comment]
yeah didn't think of that it's a good idea, I'll try to find time to submit a PR for that

**Comment by bitmold** [comment]
We made it easy to turn on always-on VPN and block connections without a VPN right from within Orbot. So closing this old ticket.
